lib/strings.nix

```diff
@@ -438,13 +438,53 @@ rec {
        => true
        isStorePath pkgs.python
        => true
+       isStorePath ./types.nix
+       => true
        isStorePath [] || isStorePath 42 || isStorePath {} || …
        => false
   */
   isStorePath = x:
-       builtins.isString x
-    && builtins.substring 0 1 (toString x) == "/"
-    && dirOf (builtins.toPath x) == builtins.storeDir;
+    let
+      toStringContext = y: "${y}";
+    in (builtins.isString x || isPath x)
+    && builtins.substring 0 1 (toStringContext x) == "/"
+    && dirOf (toStringContext x) == builtins.storeDir;
+
+  /* Check whether a string or a path is in the store.
+
+     Example:
+       isInStore "/nix/store/d945ibfx9x185xf04b890y4f9g3cbb63-python-2.7.11/bin/python"
+       => true
+       isInStore "/nix/store/d945ibfx9x185xf04b890y4f9g3cbb63-python-2.7.11/"
+       => true
+       isInStore "/home/me/whatever.nix"
+       => false
+       isInStore ./types.nix
+       => true
```

Comment:
This does not make sense, given that `toString ./types.nix` return `/local/directory/types.nix`.

Reply:
In other words, this function doesn't test whether something is in the store but whether it can be coerced to a store path in certain contexts (such as in a string anti-quotation)..

Reply:
Same could be argued that strings lose their context. It does coerce to string (and thus import it to store), but it doesn't mean it can't be converted to another type which gets another meaning.

Reply:
I don't understand...